src/ng/http.js

```diff
@@ -1279,15 +1286,18 @@ function $HttpProvider() {
       promise.then(removePendingReq, removePendingReq);
 
       if ((config.cache || defaults.cache) && config.cache !== false &&
-          (config.method === 'GET' || config.method === 'JSONP')) {
+          (isGet || isJsonp || isPostUrlencoded)) {
         cache = isObject(config.cache) ? config.cache
             : isObject(/** @type {?} */ (defaults).cache)
               ? /** @type {?} */ (defaults).cache
               : defaultCache;
+        if (isPostUrlencoded) {
+          cachedUrl = buildUrl(cachedUrl, reqData);
```

Comment:
In general, this is a string. Https://github.com/angular/angular.js/blob/master/src/ng/http.js#L1061
It's important for us to get a unique key. I can add a conversion to json, if for some reason there will be an an object